Mathlib/CategoryTheory/Bicategory/Functor/Lax.lean

```diff
@@ -0,0 +1,185 @@
+since lax functors had not yet been added (e.g `FunctorBicategory.lean`). A possible project would
+be to mirror these constructions for lax functors.
+
+-/
+
+namespace CategoryTheory
+
+open Category Bicategory
+
+open Bicategory
+
+universe w₁ w₂ w₃ v₁ v₂ v₃ u₁ u₂ u₃
+
+section
+
+variable {B : Type u₁} [Bicategory.{w₁, v₁} B] {C : Type u₂} [Bicategory.{w₂, v₂} C]
+
+/-- A lax functor `F` between bicategories `B` and `C` consists of a function between objects
+`F.obj`, a function between 1-morphisms `F.map`, and a function between 2-morphisms `F.map₂`.
+
+Unlike functors between categories, `F.map` do not need to strictly commute with the composition,
+and do not need to strictly preserve the identity. Instead, there are specified 2-morphisms
+`𝟙 (F.obj a) ⟶ F.map (𝟙 a)` and `F.map f ≫ F.map g ⟶ F.map (f ≫ g)`.
+
+`F.map₂` strictly commute with compositions and preserve the identity. They also preserve the
+associator, the left unitor, and the right unitor modulo some adjustments of domains and codomains
+of 2-morphisms.
+-/
+structure LaxFunctor (B: Type u₁) [Bicategory.{w₁, v₁} B] (C : Type u₂) [Bicategory.{w₂, v₂} C]
+    extends PrelaxFunctor B C where
+  mapId (a : B) : 𝟙 (obj a) ⟶ map (𝟙 a)
+  mapComp {a b c : B} (f : a ⟶ b) (g : b ⟶ c) : map f ≫ map g ⟶ map (f ≫ g)
+  mapComp_naturality_left :
+    ∀ {a b c : B} {f f' : a ⟶ b} (η : f ⟶ f') (g : b ⟶ c),
+      mapComp f g ≫ map₂ (η ▷ g) = map₂ η ▷ map g ≫ mapComp f' g:= by aesop_cat
+  mapComp_naturality_right :
+    ∀ {a b c : B} (f : a ⟶ b) {g g' : b ⟶ c} (η : g ⟶ g'),
+     mapComp f g ≫ map₂ (f ◁ η) = map f ◁ map₂ η ≫ mapComp f g' := by aesop_cat
+  map₂_associator :
+    ∀ {a b c d : B} (f : a ⟶ b) (g : b ⟶ c) (h : c ⟶ d),
+      -- (α_ (map f) (map g) (map h)).inv ≫ mapComp f g ▷ map h ≫ mapComp (f ≫ g) h =
+      -- map f ◁ mapComp g h ≫ mapComp f (g ≫ h) ≫ map₂ (α_ f g h).inv := by aesop_cat
+      mapComp f g ▷ map h ≫ mapComp (f ≫ g) h ≫ map₂ (α_ f g h).hom =
+      (α_ (map f) (map g) (map h)).hom ≫ map f ◁ mapComp g h ≫ mapComp f (g ≫ h) := by aesop_cat
+  map₂_leftUnitor :
+    ∀ {a b : B} (f : a ⟶ b),
+      map₂ (λ_ f).inv = (λ_ (map f)).inv ≫ mapId a ▷ map f ≫ mapComp (𝟙 a) f := by aesop_cat
+  map₂_rightUnitor :
+    ∀ {a b : B} (f : a ⟶ b),
+      map₂ (ρ_ f).inv = (ρ_ (map f)).inv ≫ map f ◁ mapId b ≫ mapComp f (𝟙 b) := by aesop_cat
+
+initialize_simps_projections LaxFunctor (+toPrelaxFunctor, -obj, -map, -map₂)
+
+namespace LaxFunctor
+
+attribute [reassoc (attr := simp)]
+  mapComp_naturality_left mapComp_naturality_right map₂_associator
+
+-- the simpNF linter complains that `map₂_leftUnitor_assoc` etc can be
+-- proved with `simp` so I move them here
+attribute [reassoc] map₂_leftUnitor map₂_rightUnitor
+attribute [simp] map₂_leftUnitor map₂_rightUnitor
+
+/-- The underlying prelax functor. -/
+add_decl_doc LaxFunctor.toPrelaxFunctor
+
+attribute [nolint docBlame] CategoryTheory.LaxFunctor.mapId
```

Comment:
At this stage it's better to have docs, at least for non-Prop fields.

Reply:
Fixed! Let me know if the description sounds reasonable, I am not an expert in bicategories.